### ISSUE TITLE
Feature request: Persist Control mappings for Remote Input and TCP (and possibly elswhere)

### ISSUE THREAD

**Issue by erict-powersoft** [melinsoftware/meos#80]
It would be nice if the control mappings for special controls, which are set up in Remote Input Server, as well as SPORTident / TCP, can be combined. I cannot imagine a use-case where they would be different.

Secondly it would also be nice if the last setup is retained for any future event - our control IDs are always the same so every race we need to set the control mappings to the same values.  These settings should be retained in Local Settings. 

Thirdly, it would be nice if the control mappings could be made available through the info server.

**Comment by erikmelin** [comment]
I agree that having them consistent and persistent within a event seems reasonable.

Saving them as a local setting I am not so sure about. If you know what you are doing its fine, but if you are not so familiar you could accidentally set something bad. Exposing them externally seems less important until there is software that will configure the SI units for a given event automatically and remotely.

**Comment by erict-powersoft** [comment]
The argument is that in our club our Start control is #11 and our Finish control is #7 – the reasons for this are lost in the mists of time.  It would be nice if that could be there the next time I start an event.  At the very least, let us set it up in (for example Remote Input Service), and then those same mappings show in SPORTident/TCP (especially since Clear is supposed to be 1, because control #1 won’t beep as long on a SIAC or SI-11 for faster clear/check – in the SPORTident manual they recommend 1 – 5 for Clear, Check, Start, Finish, Readout).

 

Thanks,

Eric

**Comment by erikmelin** [comment]
Support added in version 4.1
